csharp/src/Ice/TcpEndpoint.cs

```diff
@@ -115,25 +125,29 @@ public override void IceWritePayload(OutputStream ostr)
             bool compressionFlag)
             : base(communicator, protocol, host, port, sourceAddress, connectionId)
         {
-            Timeout = timeout;
+            Transport = transport;
             HasCompressionFlag = compressionFlag;
+            Timeout = timeout;
         }
 
-        internal TcpEndpoint(InputStream istr, Protocol protocol)
+        internal TcpEndpoint(InputStream istr, Protocol protocol, Transport transport)
             : base(istr, protocol)
         {
+            Transport = transport;
             Timeout = istr.ReadInt();
             HasCompressionFlag = istr.ReadBool();
         }
 
         internal TcpEndpoint(
             Communicator communicator,
             Protocol protocol,
+            Transport transport,
```

Comment:
Like for the factory, I would reorder: transport, protocol.

Reply:
fixed